community/bolt/src/main/java/org/neo4j/bolt/transport/NettyServer.java

```diff
@@ -109,6 +112,9 @@ public void start() throws Throwable
                                 .bind( protocolInitializer.address().socketAddress() ).sync();
                 InetSocketAddress localAddress = (InetSocketAddress) channelFuture.channel().localAddress();
                 connectionRegister.register( boltConnector.key(), localAddress );
+                ListenSocketAddress listenSocketAddress = new ListenSocketAddress(
+                        protocolInitializer.address().getHostname(), localAddress.getPort() );
+                log.info( "Bolt enabled on %s.", listenSocketAddress );
```

Comment:
Is it possible to log `localAddress` and not even create `listenSocketAddress`?

Reply:
This is wrapped in a `ListenSocketAddress` because that has extra support for formatting. The default `toString` that comes with `InetSocketAddress` is a bit messier IMHO, especially with IPv6. Plus doing this retains format consistency with what we have today.

Reply:
We could alternatively extract the formatting logic into a separate static function.

Reply:
Understood, it's just not immediately clear why is `ListenSocketAddress` required. I think it's okey to keep it as is, or call `toString()` explicitly after construction just to make it clear:

```java
String listenSocketAddress = new ListenSocketAddress(
  protocolInitializer.address().getHostname(), localAddress.getPort() ).toString();
```